tasks/snyk-code-test.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: snyk-code-test
+spec:
+  description: >-
+    Static code security test with snyk
+  results:
+    - description: Test output
+      name: HACBS_TEST_OUTPUT
+  params:
+    - name: shared-secret
+      default: test-team-snyk
+    - name: args
+      type: string
+      description: extra args needs to append
+      default: "--all-projects --exclude=test*,vendor,deps"
+  volumes:
+    - name: snyk-secret
+      csi:
+        readOnly: true
+        driver: csi.sharedresource.openshift.io
+        volumeAttributes:
+          sharedSecret: $(params.shared-secret)
+  steps:
+    - name: snyk-code-test
+      image: quay.io/redhat-appstudio/hacbs-test:feature-sast
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      volumeMounts:
+        - name: snyk-secret
+          mountPath: "/etc/secrets"
+          readOnly: true
+      script: |
+        #!/usr/bin/env bash
+        SNYK_TOKEN="$(cat /etc/secrets/snyk_token)"
+        if [[ -e $SNYK_TOKEN ]]; then
```

Comment:
`-e FILE` - FILE exists. $SNYK_TOKEN is not file, it's string. so it should be `[ -z "$SNYK_TOKEN" ]`

Reply:
Updated